---
title: "MICB425_Portfolio"
author: "Alice Sin (16528144)"
date: "version `r format(Sys.time(), '%B %d, %Y')`"
output: 
  html_document:
    toc: yes
---
# Module 01

## Data science Friday  
DS Friday Assignments are in a separate .Rmd file. 

### Installation check
Third level headers (###) should be used for links to assignments, evidence worksheets, problem sets, and readings, as seen here.  

Use this space to include your installation screenshots.



### Portfolio repo setup
```
git add .
git commit -m "input message"
git push
```


### RMarkdown pretty html challenge
It is in a separate document. 



## Origins and Earth Systems
### Evidence worksheet 01 
The template for the first Evidence Worksheet has been included here. The first thing for any assignment should link(s) to any relevant literature (which should be included as full citations in a module references section below).

You can copy-paste in the answers you recorded when working through the evidence worksheet into this portfolio template. 

As you include Evidence worksheets and Problem sets in the future, ensure that you delineate Questions/Learning Objectives/etc. by using headers that are 4th level and greater. This will still create header markings when you render (knit) the document, but will exclude these levels from the Table of Contents. That's a good thing. You don't' want to clutter the Table of Contents too much.

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

#### Learning objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems. 

#### General questions
**What were the main questions being asked?**  
How many prokaryotes are there on Earth?  
What is the abundance and location of microbes on Earth?  
Carbon content of prokaryotes inhabited in different areas?  

**What were the primary methodological approaches used?**  
- Large assumptions are made with a small sample size to estimate numbers for the entire Earth and especially for subsurface  
- Used average cell density to determine prokaryotes in the open ocean  
- Estimation using Gold: average porosity of the terrestrial surface is 3% and then assumed how much pore space each prokaryote will occupy  
- Estimation of carbon content by using number of cells in open ocean, cells, subsurface 

**Summarize the main results or findings**  
-	Prokaryotes are found in the open ocean, soil, oceanic and terrestrial subsurfaces  
-	Most of the prokaryotes on Earth’s biosphere are found in the subsurface but it is hard to calculate a precise number  
-	Prokaryotes also found in other habitats such as animals, leaves, and air but the content is much lower and impacts the total number of prokaryotes a lot less  
-	Prokaryotes are a large reservoir for essential nutrients such as nitrogen, carbon, phosphorus
o	60-100% of the carbon on earth is from prokaryotes  
o	More N & P than in plants  
-	Highest prokaryotes in air is 60-70m in the air  
-	Large amounts of autotrophs above 200m, likely to provide energy for those at lower trophic levels  
-	Mutations and rare genetic events are more likely to occur in the ocean and will use higher amount of carbon (higher turnover rate  more mutations occur)which leads to greater genetic diversity which leads to the formation of novel species so that new pathways can form for the global environment  

**Do new questions arise from the results**  
-	How do we distinguish between species of prokaryotes if there are so many genetic modifications?
-	What needs to be taken into account to distinguish prokaryotes and eukaryotes when prokaryotes have such a high turnover rate?  
-	We don’t know the exact composition of each microbes?  
-	The paper is from 1998 so are the numbers still accurate? How much has changed since then?   
- Is there better technology out there to make these calculations? Or would we still use cell density/porosity, etc?  


**Were there any specific challenges or advantages in understanding the paper (*e.g.* did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)?**  

-	Estimation/assumptions aren’t explained thoroughly  
-	Don’t show the actual calculations made so it is harder to believe  
-	Don’t know the amount of error in their calculations  only give a single number  
-	What is the uncertainty?  

### Evidence Worksheet_02 “Life and the Evolution of Earth’s Atmosphere”
#### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems
    
    
* Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago   - Solar System Formation via the Supernova Explosion
    
    + 4.2 billion years ago   - Meteorite bombardment halted; Life was habitable
    
    + 3.8 billion years ago  - Existence of Life
    
    + 3.75 billion years ago  - LHB meteorites
    
    + 3.5 billion years ago - Microfossils and stromalites present
    
    + 3.0 billion years ago  - Cyanobacteria photosynthesize
    
    + 2.7 billion years ago  - Origins of Eukarya; Great Oxidation event 
    
    + 2.2 billion years ago  - Glaciation Event and the Rise of O2
    
    + 2.1 billion years ago  - Existence of Eukaryotes
    
    + 1.3 billion years ago  - Complex Eukaryotes; multicellular life 
    
    + 550,000 years ago - Cambrian Explosion; first land plants
    
    + 200,000 years ago - Gigantism of organisms


* Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadean  
    High CO2 levels  
    Sun was not as hot
    Earth was molten rock and very hot 
    
    + Archean  
    Atmosphere had CH4 to keep Earth warm  
    Some O2 present  
    CO2 levels still high  
    
    + Precambrian  
    High atmospheric O2
    
    + Proterozoic  
    O2 and CH4 reacted to produce CO2
    There was a glaciation event
    
    + Phanerozoic  
    High atmospheric O2
    Coal deposits from organisms that died from extinction 

### Problem set 01
#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
**What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.**  

Aquatic = 1.16 x 10^24^ cells  
Soil = 2.556 x10^24^ cells   
Oceanic subsurface = 3.5 x 10^30^ cells   
Terrestrial subsurface = 0.25 - 2.5x10^30^ cells  
Total subsurface = 3.8 x 10^30^ cells  



**What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?**  

3.6x10^28 cells in the upper 200m of the ocean  
2.9x 10^27 cells are autotrophs  
Upper 200m autotrophs: 5x10^5/mL mostly Prochlorococcus (avg density 10*10^4^/mL) which is about 8% of the biomass so that means carbon is availalbe for the 92% of heterotrophs


**What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?** 

Autotroph – sustains itself using light energy found on surface of the ocean and is “self-nourishing". It can fix inorganic carbon into biomass.   

Heterotroph – it uses organic matter to produce energy  

Lithotroph – it uses inorganic substrates to produce energy  

**Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?**  

Life up until the terrestrial subsurface (3-4km). The deepest point in the ocean is Mariana's Trench (10.9km deep) so the lowest point of life could exist at 14.9km deep. The limiting factor is the temperature (125 degrees Celsius)

**Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?**  

Highest habitat: Mount Everest (8.8km above sea level)  
The primary limiting factors: lack of nutrients, lack of moisture, high amounts of ionizing radiation

**Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?**  
Top of Mount Everest (8.8km) + Mariana's Trench (10.9km) + ground surface (4km) = 24km vertical distance 

**How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)**  

```
population size * turnover/year = cells/year
```
Sample calculation
```
3.6*10^28 * 365days/16days = 8.8*10^24
```


**What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?**  

Author: assumption that carbon efficiency = 20%  
Assuming that it is 20fg of C/prokaryotic cell = 20^-30^ petagrams  

The amount of carbon in marine heterotrophs (pg/cell) = 3.6x10^28^ * 20^-30^pg of C/cell  
The amount of carbon in marine heterotrophs = 0.72  
If we use a 4 multiplier:  
4 x 0.72 = 2.88 pg of C/year  

This will vary with depth in the ocean and terrestrial and marine habitats because the amount of carbon consumed will vary depending on location and so will the microbial relationships in the population or amongst populations. 


**How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)**   

4x10^-7 mutations per DNA replication  
Want 4 of them to happen at once = 4x10^-7^ * 10^-4^ = 2.56x10^-26^  
3.6x10^28^ cells * 365/16 days = 22.8 turnovers per year  
6.8x10^24^ cels * 22.8 = 8.2 x 10^29^ cells / year  
8.2x10^29^ cells/year * 2.56x10^26^   mutations/generation = 2.1 x 10^4 mutations per year
If we convert to h per mutations: 0.4h/ point mutation 

**Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt?**   

The high mutation rate increases both the genetic diversity and adaptive potential of prokaryotes. The increased mutations means that there is a higher chance that there is an advantageous mutation to form a novel species. 

Mutations can be gained through plasmids (horizontal gene transfer), recombination, external mutagens and other methods, not just point mutations. 

**What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text?**  

The abundance of prokaryotes and its rapid replication leads to a high mutation rate. This increases its diversity in a population, which in turn increases the metabolic potential. 

    
### Problem set_02 “Microbial Engines”
#### Learning objectives: 
Discuss the role of microbial diversity and formation of coupled metabolism in driving global biogeochemical cycles.

#### Specific Questions:
**What are the primary geophysical and biogeochemical processes that create and sustain conditions for life on Earth? How do abiotic versus biotic processes vary with respect to matter and energy transformation and how are they interconnected?**  

Geophysical: Tectonic plates and atmospheric photochemical processes supply substrates to move products. These two processes allow elements and molecules to interact with each other to form and break bonds in a cycle. 


Biogeochemical: Mostly abiotically driven acid-base reactions and biologically driven redox reactions. The biogeochemical macromolecules that play a large role are H, C, N, O, S, and P. Rock weathering also drives the nutrient cycles on Earth, as it can remove CO2 and allow cellular respiration to occur. 


Abiotic processes typically use acid-base chemistry to set lower limits on external energy needed to sustain the biogeochemical cycles. It creates biogeochemical cycles at the planetary and geological time scales. In contrast, biotic processes use redox reactions and the six major macromolecules to create the average redox condition of the oceans and atmosphere. 

**Why is Earth’s redox state considered an emergent property?**  

The evolution of microbial metabolic and geochemical processes create redox conditions on Earth. All the processes are linked and interconnected. 
  

**How do reversible electron transfer reactions give rise to element and nutrient cycles at different ecological scales? What strategies do microbes use to overcome thermodynamic barriers to reversible electron flow?**  

The cycles can be oxidative or reductive and feed into new reactions to create cycles.Element and nutrient cycles arise via the reversible electron transfer When the hydrogen tension is low because the reverse process then becomes more favourable. To overcome the thermodynamic barriers of reversible electron flow, different species of microbes come together to pool their synergistic flow.

**Using information provided in the text, describe how the nitrogen cycle partitions between different redox “niches” and microbial groups. Is there a relationship between the nitrogen cycle and climate change?**  

In order for nitrogen to become accessible, N2 has to be converted to NH4+. However, nitrogenase cannot be used under aerobic conditions. This results in the partitioning of multiple microbial groups with different redox "niches". Ammonium gets converted to nitrite by Bacteria and Archaea first. Nitrite then gets converted to nitrate by chemoautotrophs. Finally, opportunistic microbes use nitrite and nitrate as electron acceptors and as a result, N2 can be used and closes the N-cycle.

There is a relationship between the nitrogen cycle and climate change. Climate change affects the level of sunlight available for the opportunistic microbes can use for photosynthesis. This affects the amount of nitrogen-fixation because it uses nitrite and nitrate as terminal electron acceptors. The nitrogen cycle also influences climate change because the nitryfing organisms may use NH4+ to reduce CO2 into organic matter, thereby reducing the greenhouse effect. 


**What is the relationship between microbial diversity and metabolic diversity and how does this relate to the discovery of new protein families from microbial community genomes?**

There is low metabolic diversity in essential multimeric microbial machines despite the fact that they may be operating sub-optimally. This results in the generation of complicated repair cycles in which new proteins may be created to repair and/or replace the sub-optimally performing protein. This conservation of essential metabolic pathways can result in new protein families arising.  

**On what basis do the authors consider microbes the guardians of metabolism?**  

The authors consider microbes the guardians of metabolism because microbes are able to protect the metabolic pathway when under environmental selection or strong environmental perturbations. In the occassion that a taxonomic unit undergoes extinction, another unit is able to keep the metabolic alive. 

### Writing assessment_01: "Microbial Engines and Humans"

  Where the final goal is to keep the human species alive, I believe that it is vital and necessary that microbes stay alive and continue to contribute to the global catalysis. This is because the products generated in the biogeochemical cycles are essential to life on this planet, which is also inclusive of humans. As indicated by our currently rapidly changing environment, we may be entering a new epoch and through history, microbes have shown us they have been able to survive through tough environmental conditions and selective pressures. It would be beneficial then, to have microbes by our side through this drastic environmental change. Although technological advances have brought us very far and has revealed a lot about how microbes function, there is still a gap in knowledge that prevents us from replicating their exact pathway, leading to our dependency on microbes. Microbial life can easily live with us, however, we cannot survive without the global catalysis and environmental transformations it provides.  
Microbes are a major contributor to important cycles which include both biogeochemical and waste cycles that are essential to human life. As the human population increases, there are increasing amounts of waste and wastewater generated in every city (1). Therefore, there needs to be an efficient method in treating wastewater that is typically rich with major nutrients within it such as nitrogen, phosphorus, carbon, and minor nutrients (1). The natural ecosystem and the natural ability for microbes to decompose, degrade, mineralize, and stabilize organic matter is a low cost and effective method as they short generation time and high metabolic rate per unit biomass in the degradation of sewage water (1). An example of treating wastewater is the transformation of organic to inorganic phosphates and taking other nutrients out of the wastewater for other uses (1). Using microbes to treat wastewater is superior as it has reduced the biological oxygen demand by 90-95% (1).  

  The resiliency of microbes to extreme environmental changes ensures its survival while humans are unable to withstand similar pressures. It is believed that a new epoch called the Anthropocene, in which humans have become the main driver of global environment change has descended upon us and that we have left the much more stable Holocene epoch where humans thrived and civilizations were able to arise (2, 3). It is believed that the Anthropocene was initiated by the beginning of the Industrial Revolution in which agriculture and industrialization grew at an accelerating rate (2). The advancement of such technologies during the industrial revolution has resulted in such a large-scale human impact on the environment that the landscape has immeasurably changed (2). This is evident through the nitrate levels of Greenland ice that have spiked unseen levels in the last 100,000 years due to the necessity of nitrogen for agricultural use (2). In addition, sediments and nonglacial organic matter has been found to contain plastics, radionuclides, pesticides, and metals due to increasing greenhouse gases that were previously unprecedented clearly draws a clear distinction from the Holocene epoch (2). It was expected that had humans not pressurized the environment through fossil fuels and agriculture, the Holoscene continue for another several thousand years (3).   
  
  The rapidly changing environment has lead to the global warming of temperatures and the continuous increase can only lead to irreversible climate change. A boundary for climate change was calculated and proposed based on the atmospheric concentration of carbon dioxide and the radiative forcing which is the amount of energy produced per unit area of the globe (3). In the case that these boundaries are breached, it is believed that we will risk irreversible climate change (3). The boundary was calculated to be 350ppm of CO2 and 1 watt per square meter and according to those parameters, our current Earth has already surpassed those numbers with 387 ppm of CO2 and 1.5 watts per metre squared change in radiative forcing (3). The retreating Arctic polar sheets and the rising sea levels is additional evidence that show support of the Earth moving into a new, warmer era, one that is outside of the stable Holocene era (3).  
  
  All of these large environmental changes are nothing new to many microbes such as cyanobacteria have been dated to survive since the Archean era, around 2.4 billion years ago (4). The existence of cyanobacteria has lead to the great oxygenation event, bringing about plants and animals, which is also inclusive of the present, oxygenated Earth humans inhabit (4). Dating back to 2.4 billion years ago shows that microbes such as cyanobacteria have been able to survive through multiple epochs with varying climates, indicating that it too is able to survive past this rapidly changing environment we are heading towards (4). If it is true that we are heading into irreversible climate changes, then microbes will be an invaluable ally and resource for humans to battle such drastic changes (3, 4).  
  
  A factor that plays into the microbes’ resilient nature to changing environments the horizontal gene flow through much of its early cellular evolution which resonates even today (5).  Entire metabolic pathways, including those that are essential to the biogeochemical cycle can be horizontally transferred to other species that may be typically responsible for other functions (5). Horizontal gene transfer is powerful in that it allows metabolic pathways to be conserved over time in the occasion that a taxonomic unit that originally had the pathway undergoes extinction (5).   
  
  In addition to the conservation of essential metabolic pathways, the environmental selection pressures on the microbes can lead to evolution of boutique, or accessory genes that can ultimately protect the metabolic pathway (5). Even in the event that a microbial machine (or pathway) functions at sub-optimal levels, they tend to be retained with minimal changes (5). Instead, new proteins will be generated or other mechanisms will be developed to support the use of the pathway (5). An example for this is the protein nitrogenase, one of the most important proteins for nitrogen-fixation that only functions in aerobic environments (5). However, most nitrogen-fixing microbes live in aerobic environments and to maintain the use of nitrogenase, organisms have protected the enzyme from oxygen by fixing nitrogen in aerobic environments (5). Horizontal gene transfer and the protective role microbes play in protecting essential metabolic pathways is evidence that in the case that microbes do have to undergo great selective pressures, as a population with multiple functions, they are able to overcome it (5).  
  
  The ability for multiple species of microbes to contribute to a larger pathway is also an advantage that gives microbes the edge over humans (5). An example would be nitrogen-fixation in which there are multiple microbes that are spatially segregated working together (5). Because nitrogenase is only able to function in anaerobic environments nitrogen-fixation was split down into parts where ammonia is initially oxidized by certain Archaea or Bacteria to nitrite and then from nitrite to nitrate by chemoautotrophic microbes, and finally, opportunistic microbes using nitrate and nitrate as electron acceptors to produce N2 (5).  
  
  Multiple species of microbes working together, combining multiple pathways in order to reach a final end product is a process that is difficult to replicate by humans due to the lack of knowledge about microbial structures, all the pathways currently used, and the proteins used (5). Although technology has advanced enough that we are able to use metagenomics to sequence full genomes of microbes, there is still not enough information for us to fully replicate what microbes do (5). This deficiency in knowledge has lead to human’s dependency on microbes. As such, we cannot live without microbes, but perhaps, as we become more knowledgeable through time, there is opportunity to use the technology we have discovered to help microbes in creating a planet that is continuously habitable in the future, despite us having possibly left the more stable Holocene epoch (3). 
  
  The global catalysis and the biogeochemical cycles of microbes are important to humans. Its role in waste cycling and its resiliency through environmental extremes shows that microbes play a strong role in our current lifestyle but was essential to introducing life on this planet (1). Its ability to retain essential metabolic pathways through many generations and use of multiple microbial species to carry out specific pathways is something that humans are currently unable to replicate as there is insufficient information on the microbes (5). There is lots of uncertainty in the future as it is hard to predict where our rapidly changing environment will take us in the future. Integrating our technological advancements in conjunction with what microbes are already doing may be a step in the forward direction. An example would be using microbes in a beneficial fashion such as extracting valuable nutrients from waste water while reducing the need for oxygen use. The levels of CO2 are only continuously rising and perhaps working with microbes may lead to a solution.   


### Module 01 references
Utilize this space to include a bibliography of any literature you want associated with this module. We recommend keeping this as the final header under each module.

An example for Whitman and Wiebe (1998) has been included below.

Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578–6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

# Module 02

### Problem set_03 “Metagenomics: Genomic Analysis of Microbial Communities”
#### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  
*(Reminders for how to format links, etc in RMarkdown are in the RMarkdown Cheat Sheets)*

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?
    - 98 bacterial phyla
    - 20 archael phyla 
    - information is gathered from small 16S rRNA databases (sequence information)
    - but there could be up to 1500 bacterial phyla because there are microbes that live in the shadow biosphere 

* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?
    - 110 000 on EBI database but it is always changing
    - Environments: ALL
        - sediments, soil, gut, aquatic
        - especially places where it's hard to culture communities

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  

    - Warehousing sequence information: INT/M, MG-RAGT (also does processing), NCBI (databases you can search that feed into other pipelines)
    - Biodiversity analysis: EstimateS
    - Microbial biodiversity analysis: MOTHUR, QIIME2
    - Binning sequences: MEGAN, CARMA  
        - use it to build a phylogenetic tree 
    
    Shotgun metagenomics
    - Analysing sequence data: FASTQC    
    - Assembly - EULER
    - Binning - GCOM
    - Annotation - KEGG
    - Analysis pipelines: MEGAN5

    Marker Gene Metagenomics
    - Standalone software: OTUbase
    - Analysis pipelines: SILVA, GreenGenes
    - Denoising: AmpliconNoise
    - Databases - Ribosomal Database Project (RDP)


* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   
    - Phylogenetic
        - vertical gene transfer 

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  

    - Metagenomics: grouping of sequences that form a single genome
    
    - Types of algorithms:
        1. Aligning sequences to databases
        2. Grouping sequences to each other based on DNA characteristics: GC content, codon usage
        
    - Risks:
        - incomplete coverage of genome sequences
        - contamination from different phylogeny 

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  

    - functional screens (ie. biochemical assays)
        - can make big librarys of DNA
    - 3rd generation sequencing (ie. Nanopore)
        - single molecule sequencing which is a good hybrid 
    - single cell sequencing 
    - FISH probe


# Module 03
### Problem set_04 “Fine-scale phylogenetic architecture”
#### Learning objectives: 

* Gain experience estimating diversity within a hypothetical microbial community


#### Outline:

In class Day 1:

1. Define and describe species within your group's "microbial" community.
2. Count and record individuals within your defined species groups.
3. Remix all species together to reform the original community.
4. Each person in your group takes a random sample of the community (*i.e.* devide up the candy).

Assignment:

5. Individually, complete a collection curve for your sample.
6. Calculate alpha-diversity based on your original total community and your individual sample.

In class Day 2:

7. Compare diversity between groups.

#### Part 1: Description and enumeration
Obtain a collection of "microbial" cells from "seawater". The cells were concentrated from different depth intervals by a marine microbiologist travelling along the Line-P transect in the northeast subarctic Pacific Ocean off the coast of Vancouver Island British Columbia. 

Sort out and identify different microbial "species" based on shared properties or traits. Record your data in this Rmarkdown using the example data as a guide.

Once you have defined your binning criteria, separate the cells using the sampling bags provided. These operational taxonomic units (OTUs) will be considered separate "species". This problem set is based on content available at [What is Biodiversity](http://cnx.org/content/m12179/latest/).

For example, load in the packages you will use.
```{r message=FALSE}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
```

Then load in the data. You should use a similar format to record your community data.  
**Sample data**
```{r}
example_data1 = data.frame(
  number = c(1,2,3, 4, 5, 6, 7, 8, 9, 10, 11),
  name = c("M&Ms", "skittles", "spheres", "strings", "jolly ranchers", "kisses", "lego", "wine gummy", "octopi gummy", "watermelon gummy", "gummy bear"),
  characteristics = c("6 different colors; chocolate inside", "5 different colours; sugar inside", "3 different colours; hard sugar", "red", "5 different colours; hard sugar", "chocolate inside", "yellow", "2 different colours", "pink and yellow; 7 legs", "sugar covered", "6 colours"),
  occurences = c(48, 52, 4, 2, 45, 7, 1, 2, 1, 1, 16)
)
```

**Community data**
```{r}
community_data = data.frame(
  number = c(1,2,3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15),
  name = c("M&Ms", "skittles", "spheres", "strings", "jolly ranchers", "kisses", "lego", "wine gummy", "octopi gummy", "cola gummies","swirl peppermint candy", "gummy bear", "sugar coated gummy bears","cherry gummy", "watermelon gummy"),
  characteristics = c("6 different colors; chocolate inside", "5 different colours; sugar inside", "3 different colours; hard sugar", "red", "5 different colours; hard sugar", "chocolate inside", "yellow", "2 different colours", "pink and yellow; 7 legs", "sugar coated", "red or blue", "6 colours", "3 colours", "sugar coated", "sugar coated"),
  occurences = c(242, 197, 19, 14, 131, 16, 17, 8, 6, 3, 3, 101, 3, 1, 2)
)
```



Finally, use these data to create a table.
**Sample data**
```{r}
example_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```

**Community data**
```{r}
community_data %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)
```
For your community:

* Construct a table listing each species, its distinguishing characteristics, the name you have given it, and the number of occurrences of the species in the collection.
* Ask yourself if your collection of microbial cells from seawater represents the actual diversity of microorganisms inhabiting waters along the Line-P transect. Were the majority of different species sampled or were many missed? 

    - Sampling bias
    - Difference in samples originally
 
#### Part 2: Collector's curve
To help answer the questions raised in Part 1, you will conduct a simple but informative analysis that is a standard practice in biodiversity surveys. This analysis involves constructing a collector's curve that plots the cumulative number of species observed along the y-axis and the cumulative number of individuals classified along the x-axis. This curve is an increasing function with a slope that will decrease as more individuals are classified and as fewer species remain to be identified. If sampling stops while the curve is still rapidly increasing then this indicates that sampling is incomplete and many species remain undetected. Alternatively, if the slope of the curve reaches zero (flattens out), sampling is likely more than adequate.

To construct the curve for your samples, choose a cell within the collection at random. This will be your first data point, such that X = 1 and Y = 1. Next, move consistently in any direction to a new cell and record whether it is different from the first. In this step X = 2, but Y may remain 1 or change to 2 if the individual represents a new species. Repeat this process until you have proceeded through all cells in your collection.

For example, we load in these data.
```{r}
example_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10,11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179),
  y = c(1,2,3,4,4,4,5,5,5,5,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,7,7,7,7,7,7,7,7,7,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8, 8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,8,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,10,10,10,10,10,10,10,10,10,10,10,10,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11,11)

)
```

And then create a plot. We will use a scatterplot (geom_point) to plot the raw data and then add a smoother to see the overall trend of the data.
```{r}
ggplot(example_data2, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

For your sample:

* Create a collector's curve for your sample (not the entire original community).
* Does the curve flatten out? If so, after how many individual cells have been collected?
    - The curve flattens out after about 150 individual cells have been collected but the slope also plateau's once in the middle of the graph at aboutr 55 individual cells.
* What can you conclude from the shape of your collector's curve as to your depth of sampling?  
    - The sample size is quite small so the graph produced is a little bit choppy. However, it does plateau at the end which shows that sampling is more than adequate and that there are not too many species missing. 

#### Part 3: Diversity estimates (alpha diversity)
Using the table from Part 1, calculate species diversity using the following indices or metrics.

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

For example, using the example data 1 with 3 species with 2, 4, and 1 individuals each, D =  

**Sample data**
```{r}
species1 = 48/(48+52+4+2+45+7+1+2+1+1+16)
species2 = 52/(48+52+4+2+45+7+1+2+1+1+16)
species3 = 4/(48+52+4+2+45+7+1+2+1+1+16)
species4 = 2/(48+52+4+2+45+7+1+2+1+1+16)
species5 = 45/(48+52+4+2+45+7+1+2+1+1+16)
species6 = 7/(48+52+4+2+45+7+1+2+1+1+16)
species7 = 1/(48+52+4+2+45+7+1+2+1+1+16)
species8 = 2/(48+52+4+2+45+7+1+2+1+1+16)
species9 = 1/(48+52+4+2+45+7+1+2+1+1+16)
species10 = 1/(48+52+4+2+45+7+1+2+1+1+16)
species11 = 16/(48+52+4+2+45+7+1+2+1+1+16)

1 / (species1^2 + species2^2 + species3^2+ species4^2+ species5^2+ species6^2+ species7^2+ species8^2+ species9^2+ species10^2+ species11^2)
```
**Community data**
```{r}
species1 = 242/806
species2 = 197/806
species3 = 19/806
species4 = 14/806
species5 = 131/806
species6 = 16/806
species7 = 17/806
species8 = 8/806
species9 = 6/806
species10 = 3/806
species11 = 3/806
species12 = 101/806
species13 = 3/806
species14 = 1/806
species15 = 2/806

1 / (species1^2 + species2^2 + species3^2+ species4^2+ species5^2+ species6^2+ species7^2+ species8^2+ species9^2+ species10^2+ species11^2+ species12^2+ species13^2+ species14^2+ species15^2)
```
The higher the value is, the greater the diversity. The maximum value is the number of species in the sample, which occurs when all species contain an equal number of individuals. Because the index reflects the number of species present (richness) and the relative proportions of each species with a community (evenness), this metric is a diveristy metric. Consider that a community can have the same number of species (equal richness) but manifest a skewed distribution in the proportion of each species (unequal evenness), which would result in different diveristy values.

* What is the Simpson Reciprocal Index for your sample? 4.350441
* What is the Simpson Reciprocal Index for your original total community? 5.15711

##### Richness: Chao1 richness estimator
Another way to calculate diversity is to estimate the number of species that are present in a sample based on the empirical data to give an upper boundary of the richness of a sample. Here, we use the Chao1 richness estimator. 

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more

So for our previous example community of 3 species with 2, 4, and 1 individuals each, $S_{chao1}$ =  
**Sample data**
```{r}
11 + 3^2/(8*2)
```

**Community data**
```{r}
15 + 1^2/(14*2)
```
* What is the chao1 estimate for your sample? 11.5625
* What is the chao1 estimate for your original total community? 15.03846

#### Part 4: Alpha-diversity functions in R
We've been doing the above calculations by hand, which is a very good exercise to aid in understanding the math behind these estimates. Not surprisingly, these same calculations can be done with R functions. Since we just have a species table, we will use the `vegan` package. You will need to install this package if you have not done so previously. 

```{r message=FALSE}
library(vegan)
```

First, we must remove the unnecesary data columns and transpose the data so that `vegan` reads it as a species table with species as columns and rows as samples (of which you only have 1).  
**Sample data**
```{r}
example_data1_diversity = 
  example_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

example_data1_diversity
```
Then we can calculate the Simpson Reciprocal Index using the `diversity` function.
```{r}
diversity(example_data1_diversity, index="invsimpson")
```
**Community data**
```{r}
community_diversity = 
  community_data %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

community_diversity
diversity(community_diversity, index="invsimpson")
```


And we can calculate the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness. This function rounds to the nearest whole number so the value will be slightly different that what you've calculated above.  
```{r}
specpool(example_data1_diversity)
```
**Community data**
```{r}
specpool(community_diversity)
```
In Project 1, you will also see functions for calculating alpha-diversity in the `phyloseq` package since we will be working with data in that form.

For your sample:

* What are the Simpson Reciprocal Indices for your sample and community using the R function?
    - Sample: 4.350441
    - Community: 4.621526
* What are the chao1 estimates for your sample and community using the R function?
    - Sample: 11
    - Community: 15
    + Verify that these values match your previous calculations.

#### Part 5: Concluding activity
If you are stuck on some of these final questions, reading the [Kunin et al. 2010](https://www.ncbi.nlm.nih.gov/pubmed/19725865) and [Lundin et al. 2012](https://www.ncbi.nlm.nih.gov/pubmed/23760801) papers may provide helpful insights. 

* How does the measure of diversity depend on the definition of species in your samples?
    - The amount of diversity increases if there are more species. This is prevalent in the Simpson Reciprocal Indices I got for my sample in comparison to the community. Because there were only 11 species in my sample compared to the 13 in the total community, the Simpson Reciprocal number was lower in my sample. 
* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?
    - I could further bin my data by separating each of these species by colour so that there could be a greater number of species. 
* How might different sequencing technologies influence observed diversity in a sample?
    - Some sequencing technologies may have higher error rates and may actually inflate the diversity estimate. However, this can be mediated by more stringent read quality filtering and lower clustering thresholds.